src/Framework/MockObject/Generator.php

```diff
@@ -944,6 +944,12 @@ private function generateMockedMethodDefinitionFromExisting($templateDir, Reflec
             $returnType = '';
         }
 
+        if (preg_match('#\*[ \t]*+@deprecated[ \t]*+(.*?)\r?+\n[ \t]*+\*(?:[ \t]*+@|/$)#s', $method->getDocComment(), $deprecation)) {
```

Comment:
I think the notice should also be triggered when there is a `@deprecated` in `$method->getDeclaringClass()->getDocComment()`.